### PR TITLE
Add configurable Quick Create mode for translations

### LINKED ISSUE #608: Add configurable Quick Create mode for translations
## Description

Add a configurable Quick Create mode for the "create new translation" button. Instead of opening `post-new.php` in a new tab, this mode creates the translation post silently via a REST API call without leaving the current post, then updates the metabox UI in place.

#606 depends on this and will continue once this is merged.

## Configuration

New checkbox in the **Advanced Settings** section (next to "Autocomplete"):

- **Label:** Quick Create
- **Description:** Create translations without leaving the current post

When disabled (default), the button behaves as before (Classic mode). When enabled, the button triggers an AJAX call instead of opening a new tab.

## Quick Create behavior

1. User clicks the "+" button for a language
2. REST API call creates a **draft** post on the target blog with:
   - Title: `From [LANG]: {original title}`
   - Content: `From [LANG]: {original content}`
   - Mapped categories and tags assigned automatically
   - MSLS link established between source and new post
3. On success, the metabox row updates in place — shows the edit link, hides the create button

## REST API endpoint

Use `register_rest_route()` under the plugin namespace instead of `admin-ajax.php`:

- **Route:** `msls/v1/create-translation` (or similar)
- **Method:** POST
- **Auth:** WordPress REST API nonce / cookie authentication
- **Parameters:** source post ID, source blog ID, target blog ID
- **Response:** new post ID, edit URL
- **Permission callback:** capability check (`edit_posts` on target blog)

## Filter hooks

Every piece of data should be filterable so developers can customize the creation pipeline:

- **Post data** — title, content, excerpt, status (before `wp_insert_post`)
- **Taxonomies** — which terms to assign, mapping logic
- **Post meta** — allow adding custom meta during creation
- **Response** — allow extending the returned data

## Scope

**Handled by default:**
- Post title and content with language prefix
- Mapped categories and tags
- MSLS link creation
- Metabox UI update

**Not handled by default, but extensible via filters:**
- Post meta and custom fields
- Custom taxonomies beyond categories/tags
- Featured image mapping
- Embedded image duplication

## Related

- #606 depends on this (currently draft, will continue after this is merged)

### PR DESCRIPTION
**Pull request by apermo**
Closes #608

## Summary

- Add **Quick Create** setting (Advanced Settings) that replaces the classic `post-new.php` link with a REST API call
- New `POST msls/v1/create-translation` endpoint creates a draft post on the target blog with mapped taxonomies and MSLS links
- Works in both the **post listing columns** and the **editor metabox** — same code path via `MslsAdminIcon::get_a()`
- All data is filterable: `msls_quick_create_post_data`, `msls_quick_create_tax_input`, `msls_quick_create_after_insert`, `msls_quick_create_response`

## Details

- Default post data: title/content prefixed with `From [lang]:`, status `draft`, mapped categories/tags
- Post meta, custom taxonomies, featured images are extensible via filters but not handled by default
- Uses semantic `<button>` element with `aria-label` for the Quick Create trigger
- On success the button is replaced with an edit link in place

## Test plan

- [ ] Enable Quick Create in Advanced Settings, verify the "+" icons in the post listing trigger AJAX creation
- [ ] Verify the created draft post has correct title, content, mapped taxonomies, and MSLS link
- [ ] Verify the icon updates to an edit link after creation
- [ ] Disable Quick Create, verify the "+" icons link to `post-new.php` as before
- [ ] Run `composer test` — 403 tests pass

### CODE REVIEW

File: includes/MslsRestApi.php [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+	 */
+	public function check_permission( \WP_REST_Request $request ): bool {
+		$target_blog_id = (int) $request->get_param( 'target_blog_id' );
+
+		switch_to_blog( $target_blog_id );
+		$can_edit = current_user_can( 'edit_posts' );
+		restore_current_blog();
+
+		return $can_edit;
+	}
```
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:84]: Good catch, also verify that this is not an existing issue in the codebase.

File: includes/MslsRestApi.php [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+	/**
+	 * @param \WP_REST_Request $request
+	 *
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	public function create_translation( \WP_REST_Request $request ) {
```
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:91]: verify the min version, and if it matches, adopt.

File: includes/MslsRestApi.php [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+		 * @param \WP_Post $source_post    The source post object.
+		 * @param int      $source_blog_id The source blog ID.
+		 * @param int      $target_blog_id The target blog ID.
+		 *
+		 * @since TBD
+		 */
+		$post_data = apply_filters( 'msls_quick_create_post_data', $post_data, $source_post, $source_blog_id, $target_blog_id );
+
+		switch_to_blog( $target_blog_id );
+		$new_post_id = wp_insert_post( $post_data, true );
```
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:137]: valid. Though also a preexisting issue if i understand it correctly

File: includes/MslsRestApi.php [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+	 * @param array<string, mixed> $post_data
+	 * @param int                  $new_post_id
+	 */
+	protected function assign_taxonomies( array $post_data, int $new_post_id ): void {
+		if ( empty( $post_data['_msls_tax_input'] ) ) {
+			return;
+		}
+
+		foreach ( $post_data['_msls_tax_input'] as $taxonomy => $term_ids ) {
+			wp_set_object_terms( $new_post_id, $term_ids, $taxonomy );
```
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:310]: default only matches tags and categories, any custom taxonomies are to be implemented via the filters, so that is likely not an issue.

File: includes/MslsRestApi.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:84]: Fixed in b7fd6a3. Added read_post capability check on source blog. ContentImporter has the same gap but is less exposed (gated behind nonce + POST).

File: includes/MslsRestApi.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:91]: Min PHP is 7.4 (composer.json), union types require 8.0. Skipped.

File: includes/MslsRestApi.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:137]: Fixed in b7fd6a3. Added post_type_exists() check before wp_insert_post(). Correct that this is a preexisting gap — ContentImporter doesn't check either.

File: includes/MslsRestApi.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:310]: Skipped — by default only mapped categories and tags are assigned. Custom taxonomies go through the msls_quick_create_tax_input filter where the implementer controls which taxonomies to include.

File: includes/MslsRestApi.php [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+		$existing_links = $source_options->get_arr();
+		restore_current_blog();
+
+		// Build a complete link map: all existing links + source + target
+		$link_map                 = $existing_links;
+		$link_map[ $source_lang ] = $source_post_id;
+		$link_map[ $target_lang ] = $new_post_id;
+
+		// Update every blog in the link map
+		foreach ( $link_map as $lang => $post_id ) {
```
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:342]: Fixed in b7fd6a3. Added early continue for empty post_id values in the link map loop.

File: includes/MslsRestApi.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:84]: https://github.com/lloc/Multisite-Language-Switcher/issues/610

File: includes/MslsRestApi.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (apermo) [reply includes/MslsRestApi.php:137]: https://github.com/lloc/Multisite-Language-Switcher/issues/611

**Review by lloc — APPROVED** [review]

**Comment by lloc** [comment]
Thanks for working on this!
